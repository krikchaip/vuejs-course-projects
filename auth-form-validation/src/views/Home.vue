<template>
  <div class="jumbo">
    <div class="jumbo__heading">Time to learn something new!</div>
    <div class="jumbo__text">
      You found the best place to learn -
      and now you're just one sign in (or sign up) away from it!
    </div>
    <div class="jumbo__actions">
      <router-link class="jumbo__btn btn" to="/signup">Sign Up</router-link>
      <router-link class="jumbo__btn btn" to="/signin">Sign In</router-link>
    </div>
  </div>
</template>
