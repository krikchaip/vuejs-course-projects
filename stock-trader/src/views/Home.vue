<template>
  <div class="card">
    <div class="card-body">
      <div class="h2">Trade or View your Portfolio</div>
      <div class="text-muted">You may Save & Load your Data</div>
      <div class="text-muted">Click on "End Day" to begin a new Day!</div>
      <hr>
      <strong>Your Funds: ${{ (10000).toLocaleString() }}</strong>
    </div>
  </div>
</template>
