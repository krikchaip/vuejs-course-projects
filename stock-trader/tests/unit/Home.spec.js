import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'

describe('funds getter is set', () => {
  const $store = {
    getters: { funds: '100,000' }
  }

  it('should display correct funds with messages', () => {
    const wrapper = shallowMount(Home, { mocks: { $store } })
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.vm.$store.getters.funds = '9,000'
    expect(wrapper.html()).toMatchSnapshot()
  })
})
