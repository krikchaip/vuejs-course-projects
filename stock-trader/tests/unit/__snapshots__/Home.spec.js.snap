// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`funds getter is set should display correct funds with messages 1`] = `
<div class="card p-4">
  <h1 class="h2">Trade or View your Portfolio</h1>
  <small>You may Save &amp; Load your Data</small>
  <small>Click on "End Day" to begin a new Day!</small>
  <hr> <strong>Your Funds: $100,000</strong></div>
`;

exports[`funds getter is set should display correct funds with messages 2`] = `
<div class="card p-4">
  <h1 class="h2">Trade or View your Portfolio</h1>
  <small>You may Save &amp; Load your Data</small>
  <small>Click on "End Day" to begin a new Day!</small>
  <hr> <strong>Your Funds: $9,000</strong></div>
`;
